test/functional_test.go

```diff
@@ -70,9 +72,12 @@ func TestFunctional(t *testing.T) {
 	binary := CompileBinary(t, projectDir, tempDir)
 
 	// Run test for each directory
-	for _, testDir := range GetTestDirs(t, rootDir) {
+	for _, d := range GetTestDirs(t, rootDir) {
+		testDir := d
```

Comment:
[Be Careful with Table Driven Tests and t.Parallel()](https://gist.github.com/posener/92a55c4cd441fc5e5e85f27bca008721)